tests/end-to-end/backup-restore-test/backupe2e/grafana.go

```diff
@@ -0,0 +1,372 @@
+	So(err, ShouldBeNil)
+
+	deletePolicy := metav1.DeletePropagationForeground
+
+	serviceList, err := t.coreClient.CoreV1().Services(grafanaNS).List(metav1.ListOptions{LabelSelector: grafanaLabelSelector,})
+	So(err, ShouldBeNil)
+
+	for _, service := range serviceList.Items {
+		if service.Name == grafanaServiceName {
+			// Delete Service
+			err = t.coreClient.CoreV1().Services(grafanaNS).Delete(grafanaServiceName, &metav1.DeleteOptions{
+				PropagationPolicy: &deletePolicy,
+			})
+			So(err, ShouldBeNil)
+		}
+	}
+
+	collection := t.coreClient.AppsV1().StatefulSets(grafanaNS)
+	err = collection.Delete(grafanaStatefulsetName, &metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	})
+	So(err, ShouldBeNil)
+
+	podList, err := t.coreClient.CoreV1().Pods(grafanaNS).List(metav1.ListOptions{LabelSelector: grafanaLabelSelector,})
+	So(err, ShouldBeNil)
+
+	for _, pod := range podList.Items {
+		if pod.Name == grafanaPodName {
+			// Delete Pod
+			err = t.coreClient.CoreV1().Pods(grafanaNS).Delete(grafanaPodName, &metav1.DeleteOptions{})
+			So(err, ShouldBeNil)
+		}
+	}
+
+	err = t.waitForPodGrafana(2 * time.Minute)
+	So(err, ShouldBeError)
+
+}
+
+func (t *grafanaTest) TestResources(namespace string) {
+	err := t.waitForPodGrafana(5 * time.Minute)
+	So(err, ShouldBeNil)
+
+	err = t.getGrafana()
+	So(err, ShouldBeNil)
+
+	dexAuthLocal := t.getGrafanaAndDexAuth()
+
+	// request for retrieving folders and dashboards of the general folder
+	// /api/search?folderIds=0
+	domain := fmt.Sprintf("%s%s", dexAuthLocal.Request.URL.String(), "api/search")
+	params := url.Values{}
+	params.Set("folderIds", "0")
+	cookie := dexAuthLocal.Request.Cookies()
+	apiSearchFolders, err := t.requestToGrafana(domain, "GET", params, nil, cookie)
+	So(err, ShouldBeNil)
+	So(apiSearchFolders.StatusCode, ShouldEqual, http.StatusOK)
+
+	defer apiSearchFolders.Body.Close()
+	dataBody, err := ioutil.ReadAll(apiSearchFolders.Body)
+	So(err, ShouldBeNil)
+
+	// Query to api
+	dashboardFolders := make([]map[string]interface{}, 0)
+	err = json.Unmarshal(dataBody, &dashboardFolders)
+	So(err, ShouldBeNil)
+	So(len(dashboardFolders), ShouldNotEqual, 0)
+
+	for _, folder := range dashboardFolders {
+		// http request to every dashboard
+		domain = fmt.Sprintf("%s%s", t.url, folder["url"])
+		formData := url.Values{}
+		formData.Set("login", t.user)
+		formData.Set("password", t.password)
+		dashboard, err := t.requestToGrafana(domain, "GET", nil, strings.NewReader(formData.Encode()), cookie)
+		So(err, ShouldBeNil)
+		So(dashboard.StatusCode, ShouldEqual, http.StatusOK)
+		title := folder["title"].(string)
+		switch title {
+		case "Lambda Dashboard":
+			apiRequestsStatus[title] = dashboard.StatusCode
+		case "Etcd":
+			apiRequestsStatus[title] = dashboard.StatusCode
+		case "Connector-Service":
+			apiRequestsStatus[title] = dashboard.StatusCode
+		case "Istio Mesh Dashboard":
+			apiRequestsStatus[title] = dashboard.StatusCode
+		case "Istio Service Dashboard":
+
+		}
```

Comment:
you could do the following:
before backup: query all dashboard, and store list in test object, try to access all of those dashboards
after backup: dont query the list of dashboard, but just use the list from the previous run, try to access all of them again